File: python/src/rag/chunking.py
def chunk_text(text: str, chunk_size: int = 800, overlap: int = 150) -> list[str]:
    cleaned = " ".join(text.split())
    if not cleaned:
        return []

    chunks: list[str] = []
    start = 0
    length = len(cleaned)

    while start < length:
        end = min(start + chunk_size, length)
        chunk = cleaned[start:end]
        chunks.append(chunk)
        if end >= length:
            break
        start = max(0, end - overlap)

    return chunks

File: node/src/rag/chunking.mjs
// chunkText splits text into overlapping chunks for embedding.
export function chunkText(text, chunkSize = 800, overlap = 150) {
  const cleaned = text.split(/\s+/).filter(Boolean).join(" ");
  if (!cleaned) {
    return [];
  }

  const chunks = [];
  let start = 0;
  while (start < cleaned.length) {
    const end = Math.min(start + chunkSize, cleaned.length);
    chunks.push(cleaned.slice(start, end));
    if (end >= cleaned.length) {
      break;
    }
    start = Math.max(0, end - overlap);
  }

  return chunks;
}